dependencies.yaml

```diff
@@ -165,10 +174,21 @@ dependencies:
     specific:
       - output_types: conda
         matrices:
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
+              - cuda-cudart-dev
+              - cuda-profiler-api
+              - libcublas-dev
+              - libcurand-dev
+              - libcusolver-dev
+              - libcusparse-dev
```

Comment:
From what I can see of where the `cudatoolkit` list is used, most of the places are runtime rather than build envs. Should we be including the non `-dev` packages instead?

And if we do need the dev packages, do we need `cuda-nvcc` @bdice?

Reply:
I think you're correct and we want to use the non `dev` packages, since it is a runtime dependency and not part of the API and users don't need to use the headers of this libraries directly

Reply:
(Not at a computer right now so pardon the brevity)

This is intentional. RAFT is a special case within RAPIDS because it is header-only (like RMM) and has nontrivial library dev dependencies (unlike RMM). We must have the dev packages to build RAFT, but we also list these dev packages as run dependencies for libraft-headers (but not libraft-headers-only) because they are required to use the headers. We can discuss the pros and cons of whether header-only libraries should list their dev dependencies in run (this is a choice that can go either way), but I would prefer to be as consistent as possible with past decisions for CUDA 11 and the current conda recipes for this PR, and most importantly avoid holding back merging for packaging discussions if we can defer them (to prevent blocking cuml/cugraph from starting their CUDA 12 work).

Reply:
Oh of course wasn't thinking about the header-only bit. OK yeah that makes sense. But then practically speaking we are always going to need cuda-nvcc as well. I'm fine not listing it and leaving that to the user since it's the compiler if that's your preference.